LibraBFT/Impl/Consensus/RoundManager/Properties.agda

```diff
@@ -123,69 +141,155 @@ module processProposalMSpec (proposal : Block) where
-            let vm = VoteMsg∙new vote si in
-            mkContract (Right (mkVoteMsgOutsCorrect vm recipient
-                                (++-NoMsgOuts-VoteMsgOuts outs (SendVote vm (recipient ∷ []) ∷ []) vm (recipient ∷ [])
-                                   noOuts refl)
-                                (substVoteCorrect refl refl refl refl refl refl resultCorrect)))
+    contractBail{outs} nmo =
+      mkContract
+        reflNoEpochChange
+        (NoMsgOuts⇒NoBroadcastOuts{outs} nmo)
+        (Left (true , mkNoVoteMsgOutsCorrect (NoMsgOuts⇒NoVoteOuts{outs} nmo) (Left reflNoVoteCorrect)))
+
+    contract-step₂ : RWST-weakestPre (executeAndVoteM proposal >>= step₂) (Contract pre) unit pre
+    contract-step₂ =
+      executeAndVoteMSpec.contract proposal pre
+        (RWST-weakestPre-bindPost unit step₂ (Contract pre)) pf
+      where
+      pf : RWST-Post-⇒ (executeAndVoteMSpec.Contract proposal pre) (RWST-weakestPre-bindPost unit step₂ (Contract pre))
+      pf (Left x) st outs (executeAndVoteMSpec.mkContract noOuts inv lvrStrict? resultCorrect) ._ refl =
+        mkContract inv
+          (++-NoMsgOuts-NoBroadcastOuts outs (LogErr fakeErr ∷ []) noOuts refl)
+          (Left (lvrStrict? , mkNoVoteMsgOutsCorrect (++-NoMsgOuts-NoVoteOuts outs _ noOuts refl) resultCorrect))
+      pf (Right vote) st outs (executeAndVoteMSpec.mkContract noOuts inv lvrStrict? resultCorrect) ._ refl ._ refl ._ refl =
+        syncInfoMSpec.contract (st & rsVoteSent-rm ∙~ just vote)
+          (RWST-weakestPre-bindPost unit (step₃ vote) (RWST-Post++ (Contract pre) outs))
+          λ si si≡ → contract-step₃ si
+        where
+        module _ (si : SyncInfo) where
+          contract-step₃ : RWST-weakestPre (step₃ vote si) (RWST-Post++ (Contract pre) outs) unit (st & rsVoteSent-rm ∙~ just vote)
+          contract-step₃ ._ refl ._ refl ._ refl ._ refl recipient@._ refl =
+            mkContract
               (transNoEpochChange inv (mkNoEpochChange refl refl))
+              (++-NoMsgOuts-NoBroadcastOuts outs _ noOuts refl)
+              (Right (mkVoteMsgOutsCorrect
+                       (VoteMsg∙new vote si) recipient
+                       (++-NoMsgOuts-VoteMsgOuts outs _ ((VoteMsg∙new vote si)) (recipient ∷ []) noOuts refl)
+                       (substVoteCorrect refl refl refl refl refl refl resultCorrect)))
+
+  contract : ∀ pre Post → RWST-Post-⇒ (Contract pre) Post → LBFT-weakestPre (processProposalM proposal) Post pre
+  contract pre Post pf = LBFT-⇒ (Contract pre) Post pf (processProposalM proposal) pre (contract' pre)
+
+module syncUpM
+  (now : Instant) (syncInfo : SyncInfo) (author : Author) (_helpRemote : Bool) where
+
+  record Contract (pre : RoundManager) (r : Either ErrLog Unit) (post : RoundManager) (outs : List Output) : Set where
+    constructor mkContract
+    field
+      noEpochChange : NoEpochChange pre post
+      noVoteOuts    : NoVoteOuts outs
+      noVote        : NoVoteCorrect pre post true
+
+  postulate
```

Comment:
According to our [style guide](https://github.com/oracle/bft-consensus-agda/blob/main/STYLE.md#assumptions-warnings-and-overrides), there should be a comment after each postulate, with either `TODO-n` or a note that it's a valid assumption, etc.  These should start on the same line as the postulate for easy grepping.

In addition, where we're postulating contracts that will need to be proved, it would be useful to give some indication about things like:
* Whether the property is ready to be proved or (for example) requires refinement first.
* How easy/hard this one is expected to be.
* Any expected challenges, hints, etc.

Maybe we should consider a Beginner/Intermediate/Advanced kind of estimation?